src/controller/java/BUILD.gn

```diff
@@ -122,12 +122,47 @@ android_library("java") {
     "zap-generated/chip/devicecontroller/ClusterWriteMapping.java",
   ]
 
+  if (build_java_matter_controller) {
+    sources += [
+          "stub_src/android/annotations/Nullable.java",
+          "stub_src/android/bluetooth/BluetoothGatt.java",
+          "stub_src/android/bluetooth/BluetoothGattCallback.java",
+          "stub_src/android/bluetooth/BluetoothGattCharacteristic.java",
+          "stub_src/android/bluetooth/BluetoothGattDescriptor.java",
+          "stub_src/android/bluetooth/BluetoothGattService.java",
+          "stub_src/android/bluetooth/BluetoothProfile.java",
+          "stub_src/android/os/Build.java",
+          "stub_src/android/os/Parcelable.java",
+          "stub_src/android/os/Parcel.java",
+          "stub_src/android/util/Log.java",
+    ]
+    deps = [
+      ":json",
+    ]
+  }
+  else
+  {
+    deps = [
+      ":android",
+      "${chip_root}/third_party/android_deps:annotation",
+    ]
+  }
+
   javac_flags = [ "-Xlint:deprecation" ]
 
   # TODO: add classpath support (we likely need to add something like
   #  ..../platforms/android-21/android.jar to access BLE items)
 }
 
-java_prebuilt("android") {
-  jar_path = "${android_sdk_root}/platforms/android-21/android.jar"
+if (build_java_matter_controller) {
+  java_prebuilt("json") {
```

Comment:
if we use JDK instead of Android SDK to build java controller, we have json lib to use, right? no need to include the pre-build json lib

Reply:
we have multiple popular json libraries, for example, JSON-Java, GSON, json-simple, they have different interface/API. Currently android sdk provides json-java for the use in our java code, we need to download this lib from maven, and put it in third_party, and use prebuilt java template to include it.
yes, we need include the prebuild json lib